proxy/proxy.go

```diff
@@ -1193,9 +1193,7 @@ func (p *Proxy) do(ctx *context, parentSpan ot.Span) (err error) {
 		if err := p.do(loopCTX, loopSpan); err != nil {
 			// in case of error we have to copy the response in this recursion unwinding
 			ctx.response = loopCTX.response
-			if err != nil {
```

Comment:
not related to PR but decided to fix to remove linter warning, the if condition ensures we have error so the inner one is redundant

Reply:
Okay, I see outer `err != nil` on the line 1193, I agree.